src/System.Net.Http/src/System/Net/Http/Headers/ObjectCollection.cs

```diff
@@ -23,6 +24,13 @@ public ObjectCollection(Action<T> validator)
             _validator = validator;
         }
 
+        // This is only used internally to enumerate the collection
+        // without the enumerator allocation.
+        new public List<T>.Enumerator GetEnumerator()
+        {
+            return ((List<T>)Items).GetEnumerator();
```

Comment:
This assumes that Items is in fact a ```List<T>```, which depends on an implementation detail of ```Collection<T>``` and the ctor of ```ObjectCollection<T>``` not passing in a different list type.  You might consider modifying the ```ObjectCollection(Action<T> validator)``` ctor to be ```ObjectCollection(Action<T> validator) : base(new List<T>())``` to make that explicit.